### PR TITLE
Fix JWT key validation

### PR DESCRIPTION
**Pull request by micahmo**
## Pull request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] New feature or enhancement
- [ ] UI change (please include screenshot!)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Internationalization and localization
- [ ] Other (please describe):

## What is the current behavior?

Unless I'm missing something obvious, it's currently impossible to validate tokens with a public key due to two issues:
 - The token algorithm is never set, so the key-building code always tries the wrong algorithm.
 - The public key text box is never displayed, because the visibility is only changed after a successful validation, which can't happen without the public key (chicken-and-egg).

(The unit tests have worked because they set the algorithm and public key explicitly.)

I also fixed a small issue related to **en**coding where the error banner would show when there wasn't an error and hide when there was!

## What is the new behavior?

**Before**: Don't even see the option for a public key. Obviously pasting it in the signature field doesn't work.

https://github.com/veler/DevToys/assets/7417301/4cc472c0-f78c-46fd-9703-0f86886ade93

**After**: Pasting a JWT with an RSA-encoded signature shows the Public key field and successfully validates.

https://github.com/veler/DevToys/assets/7417301/d1a93624-b1ef-4f82-9aa4-3e1b47d4d096

## Quality check

Before creating this PR:

- [x] Did you follow the code style guideline as described in [CONTRIBUTING.md](https://github.com/veler/DevToys/blob/main/CONTRIBUTING.md)
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? On Windows, you can use Accessibility Insights for this.
- [ ] Did you verify that the change work in Release build configuration
- [x] Did you verify that all unit tests pass
- [ ] If necessary and if possible, did you verify your changes on:
   - [x] Windows
   - [ ] macOS (DevToys 2.0)
   - [ ] Linux (DevToys 2.0)

### CODE REVIEW

**Review by btiteux — APPROVED** [review]
Hello, thank you for this fix, I totally forgot the public key validation.

File: src/dev/impl/DevToys/ViewModels/Tools/EncodersDecoders/JwtDecoderEncoder/JwtDecoder.cs [hunk 1]
```diff
@@ -33,12 +33,13 @@ public class JwtDecoder
         public TokenResult? DecodeToken(
                 DecoderParameters decodeParameters,
                 TokenParameters tokenParameters,
-                Action<TokenResultErrorEventArgs> decodingErrorCallBack)
+                Action<TokenResultErrorEventArgs> decodingErrorCallBack, out JwtAlgorithm? jwtAlgorithm)
```
Comment (veler) [inline src/dev/impl/DevToys/ViewModels/Tools/EncodersDecoders/JwtDecoderEncoder/JwtDecoder.cs:36]: Can you please bring this last parameter to another line? It helps readability.

**Comment by veler** [comment]
I see you updated the unit tests, but I don't seem to find any added unit tests that covers this scenario. Can we please add some?

File: src/dev/impl/DevToys/ViewModels/Tools/EncodersDecoders/JwtDecoderEncoder/JwtDecoder.cs (hunk 1 above)
Comment (micahmo) [reply src/dev/impl/DevToys/ViewModels/Tools/EncodersDecoders/JwtDecoderEncoder/JwtDecoder.cs]: Done! 6d8dfd0c3a7a2068bbe1f0d1e2ced62dee02c7fc

**Comment by micahmo** [comment]
> I see you updated the unit tests, but I don't seem to find any added unit tests that covers this scenario. Can we please add some?

Yes, I only updated the tests as needed to match the new signature of `DecodeToken`. But I agree, I should validate that the method properly detects and assigns the algorithm rather than specifying it in the test input. Done in 029433303d304d6ff738988b32e561816c90460e.

**Pull request merged**